third_party/move/move-compiler-v2/src/function_checker.rs

```diff
@@ -0,0 +1,246 @@
+                                },
+                                Visibility::Private => {
+                                    if caller_is_inline || callee_is_inline {
+                                        let call_details: Vec<_> = sites
+                                            .iter()
+                                            .map(|node_id| {
+                                                (
+                                                    env.get_node_loc(*node_id),
+                                                    "called here".to_owned(),
+                                                )
+                                            })
+                                            .collect();
+                                        let msg = format!(
+                                            "{}function `{}` cannot be called from {}function `{}` \
+                                             because it is private to module `{}`",
+                                            if callee_is_inline { "inline " } else { "" },
+                                            callee_env.get_full_name_with_address(),
+                                            if caller_is_inline { "inline " } else { "" },
+                                            caller_name_str,
+                                            callee_env.module_env.get_full_name_str());
+                                        env.diag_with_labels(
+                                            Severity::Error,
+                                            &callee_env.get_loc(),
+                                            &msg,
+                                            call_details,
+                                        );
+                                    }
+                                    false
+                                },
+                            }
+                        };
+                        if callee_is_accessible {
+                            functions_with_callers.insert(callee);
+                        } else {
+                            functions_with_inaccessible_callers.insert(callee);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // Check for Unused functions: private (or friendless public(friend)) funs with no callers.
+    for callee in private_funcs {
+        if !functions_with_callers.contains(&callee) {
+            // We saw no uses of private/friendless function `callee`.
+            let callee_env = env.get_function(callee);
+            let callee_loc = callee_env.get_loc();
+            let callee_is_script = callee_env.module_env.get_name().is_script();
+
+            // Entry functions in a script don't need any uses.
+            // Check others which are private.
+            if !callee_is_script {
+                let is_private = matches!(callee_env.visibility(), Visibility::Private);
+                if warn_about_unused {
```

Comment:
Move this check just outside the enclosing for loop?

Reply:
It can go outside the outer loop, too.